MetaMask/metamask-extension
# fix: upgrade transaction-controller to 32.0.0 to fix mmi

> [!NOTE]
> This PR is intended to be cherry-picked into RC 11.17.0 after being merged into develop

## **Description**
Upgrade transaction-controller to get this fix: https://github.com/MetaMask/core/pull/4343

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24947?quickstart=1)

## **Related issues**

Fixes:
A recent update to the transaction-controller has made the TransactionMeta object passed to the afterSign hook frozen. This change prevents adding new properties, leading to the error: “Cannot add property custodyId, object is not extensible.” This bug is breaking all transactions for MMI as the original txMeta cannot store required properties like custodyId.

Blocked by https://github.com/MetaMask/metamask-extension/pull/24861 
Blocked by https://github.com/MetaMask/metamask-extension/pull/24913

## **Manual testing steps**
I followed these steps to test the fix:
1. yarn && yarn start:mmi
2. create a new wallet
3. visit https://saturn-custody-ui.dev.metamask-institutional.io/ and add two custodial addresses
4. send 0ETH from one custodial address to the other

You should see a popup with an Approve button. Before this fix, the transaction would appear in the activity tab with an error.


## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
